apps/dashboard/src/hooks/use-subscribers-url-state.ts

```diff
@@ -0,0 +1,90 @@
+import { useCallback, useMemo } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { SubscribersFiltersData } from '../components/subscribers/subscribers-filters';
+import { useDebounce } from '../hooks/use-debounce';
+
+interface SubscribersFilters {
+  email?: string;
+  phone?: string;
+  name?: string;
+  subscriberId?: string;
+}
+
+export interface SubscribersUrlState {
```

Comment:
I am not a big fan of this modeling for the state. I think a Map or a simple POJO will work better, especially when we have to add multiple filtering values of the same type in the future. For example, filter subscribers by email1, email2, etc...

Reply:
I followed what we did for the activity feed here, but we should change it when we need more? It'd be good to create sth reusable though.